pallene/checker.lua

```diff
@@ -507,18 +522,17 @@ function Checker:check_stat(stat, is_toplevel)
     elseif tag == "ast.Stat.Assign" then
         self:expand_function_returns(stat.exps)
 
-        local declarations = {}
-
         for i, var in ipairs(stat.vars) do
             if var._tag == "ast.Var.Dot" and self:is_the_module_variable(var.exp) then
                 -- Declaring a module field
                 if not is_toplevel then
                     type_error(var.loc, "module fields can only be set at the toplevel")
                 end
-                if self.module_symbol.symbols[var.name] or declarations[var.name] then
-                    multiple_definitions_error(var.loc, var.name)
-                end
-                declarations[var.name] = i
+                local qvar = ast.Var.Name(var.loc, var.name)
```

Comment:
What does `qvar` mean?

Reply:
That `qvar` name is referencing something that no longer exists. I'll change it.

When I was working on that checker refactoring that introduced the Defs, I experimented with having a new kind of `Var` called `ast.Var.QualifiedName`. However, I ended up not doing that, because the `Def` already served that purpose.